schema/migration-2-0014-20200430.sql

```diff
@@ -0,0 +1,320 @@
+-- Persistent generated migration.
+
+CREATE FUNCTION migrate() RETURNS void AS $$
+DECLARE
+  next_version int ;
+BEGIN
+  SELECT stage_two + 1 INTO next_version FROM schema_version ;
+  IF next_version = 14 THEN
+
+    -- TX.
+
+    -- BLOCK.
+    -- For Shelley. All of them can be NULLABLE since we have one table for both
+    -- Byron and Shelley.
+    EXECUTE 'ALTER TABLE block ADD COLUMN "block_issuer" hash32type NULL';
+    EXECUTE 'ALTER TABLE block ADD COLUMN "vrf_key" hash32type NULL';
+    EXECUTE 'ALTER TABLE block ADD COLUMN "nonce_vrf" hash32type NULL';
+    EXECUTE 'ALTER TABLE block ADD COLUMN "leader_vrf" hash32type NULL';
+    EXECUTE 'ALTER TABLE block ADD COLUMN "op_cert" hash32type NULL';
+    EXECUTE 'ALTER TABLE block ADD COLUMN "proto_version" hash32type NULL';
+
+    -- TXIN.
+    -- One diff, the index.
+    
+    -- TXOUT.
+    
+    -- NEW for SHELLEY
+
+    -- CENTRAL_FUNDS
+    -- DO we connect up with the EPOCH?
+    -- FOREIGN KEY (epoch_no) REFERENCES epoch (id)
+
+    CREATE TABLE central_funds (
+        id                      bigint              NOT NULL,
+        epoch_no                uinteger            NOT NULL,
+        treasury                lovelace            NOT NULL,
+        reserves                lovelace            NOT NULL,
+
+        PRIMARY KEY (id),
+        CONSTRAINT central_funds_uk UNIQUE (epoch_no)
+    );
+    
+
+    CREATE SEQUENCE public.central_funds_id_seq
+        START WITH 1
+        INCREMENT BY 1
+        NO MINVALUE
+        NO MAXVALUE
+        CACHE 1;
+    
+    ALTER SEQUENCE public.central_funds_id_seq OWNED BY public.central_funds.id;
+
+    -- Basic rewards data.
+    -- What reward has been earned in the epoch by delegating to the specified pool id.
+    -- This design allows rewards to be discriminated based on how they are earned.
+
+    CREATE TABLE reward (
+        id                      bigint              NOT NULL,
+        reward_address          staking_addr_hash   NOT NULL,
+        instantaneous           boolean             NOT NULL,
+        pool_id                 hash32type          NOT NULL,  -- Arbitrary if an instantaneous reward - needed for primary key
```

Comment:
Were these comments added manually? Probably not a good idea to manually add comments to a generated file.

The comments belong in the Haskell schema definition.

Reply:
This whole file was added manually.

Reply:
It was added to git manually, but wasn't generated using cardano-db-tool create-migration` ?

If so, deleting it and recreating it with `create-migration` should produce the same file.

Reply:
The other thing is that we are likely to want to collapse all the migrations periodically.

Comments in these generated files is unlikely to survive migration collapsing. Comments should go in the Persistent schema definition.

Reply:
Regenerated the file.